modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/human.dm

```diff
@@ -48,111 +48,87 @@
-				if(peepee.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless())
-					return TRUE
-				else
-					return FALSE
-			else
-				return TRUE
-	return FALSE
+	var/obj/item/organ/external/genital/genital = getorganslot(ORGAN_SLOT_PENIS)
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return genital.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless()
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			return genital.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless()
+		else
+			return TRUE
 
+/// Returns true if the human has a accessible balls for the parameter. Accepts any of the `REQUIRE_GENITAL_` defines.
 /mob/living/carbon/human/proc/has_balls(required_state = REQUIRE_GENITAL_ANY)
-	var/obj/item/organ/external/genital/peepee = getorganslot(ORGAN_SLOT_TESTICLES)
-	if(peepee)
-		switch(required_state)
-			if(REQUIRE_GENITAL_ANY)
-				return TRUE
-			if(REQUIRE_GENITAL_EXPOSED)
-				if(peepee.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless())
-					return TRUE
-				else
-					return FALSE
-			if(REQUIRE_GENITAL_UNEXPOSED)
-				if(peepee.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless())
-					return TRUE
-				else
-					return FALSE
-			else
-				return TRUE
-	return FALSE
+	var/obj/item/organ/external/genital/genital = getorganslot(ORGAN_SLOT_TESTICLES)
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return genital.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless()
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			return genital.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless()
+		else
+			return TRUE
 
+/// Returns true if the human has an accessible vagina for the parameter. Accepts any of the `REQUIRE_GENITAL_` defines.
 /mob/living/carbon/human/proc/has_vagina(required_state = REQUIRE_GENITAL_ANY)
-	if(issilicon(src) && has_vagina)
-		return TRUE
-	var/obj/item/organ/external/genital/peepee = getorganslot(ORGAN_SLOT_VAGINA)
-	if(peepee)
-		switch(required_state)
-			if(REQUIRE_GENITAL_ANY)
+	var/obj/item/organ/external/genital/genital = getorganslot(ORGAN_SLOT_VAGINA)
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return genital.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless()
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			if(genital.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless())
 				return TRUE
-			if(REQUIRE_GENITAL_EXPOSED)
-				if(peepee.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless())
-					return TRUE
-				else
-					return FALSE
-			if(REQUIRE_GENITAL_UNEXPOSED)
-				if(peepee.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless())
-					return TRUE
-				else
-					return FALSE
 			else
-				return TRUE
-	return FALSE
+				return FALSE
```

Comment:
Just do `return genital.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless()` here instead of an if()

Reply:
Oops, missed that one!